src/olympia/blocklist/mlbf.py

```diff
@@ -0,0 +1,84 @@
+import math
+import secrets
+
+from filtercascade import FilterCascade
+
+import olympia.core.logger
+
+log = olympia.core.logger.getLogger('z.amo.blocklist')
+
+
+def get_blocked_guids():
+    from olympia.addons.models import Addon
+    from olympia.blocklist.models import Block
+
+    blocks = Block.objects.all()
+    blocks_guids = [block.guid for block in blocks]
+    addons_dict = Addon.unfiltered.in_bulk(blocks_guids, field_name='guid')
+    for block in blocks:
+        block.addon = addons_dict.get(block.guid)
+    Block.preload_addon_versions(blocks)
+    all_versions = {}
+    # collect all the blocked versions
+    for block in blocks:
+        is_all_versions = (
+            block.min_version == Block.MIN and
+            block.max_version == Block.MAX)
+        versions = {
+            version_id: (block.guid, version)
+            for version, (version_id, _) in block.addon_versions.items()
+            if is_all_versions or block.is_version_blocked(version)}
+        all_versions.update(versions)
+    return all_versions.values()
+
+
+def get_all_guids():
+    from olympia.versions.models import Version
+
+    return Version.objects.values_list('addon__guid', 'version')
```

Comment:
You want deleted versions as well, I think. So `Version.unfiltered`.

Reply:
https://github.com/mozilla/addons-server/pull/13659/commits/d34fd01b586cafd8a53855d9b5dc1c1e9056074a